st2common/st2common/services/packs.py

```diff
@@ -215,3 +221,68 @@ def search_pack_index(
                 break
 
     return list(itertools.chain.from_iterable(matches))
+
+
+def delete_action_files_from_pack(pack_name, entry_point, metadata_file):
+    """
+    Prepares the path for entry_point file and metadata file of action and
+    deletes them from disk.
+    """
+    pack_base_path = get_pack_base_path(pack_name=pack_name)
+    action_entrypoint_file_path = os.path.join(pack_base_path, "actions", entry_point)
+    action_metadata_file_path = os.path.join(pack_base_path, metadata_file)
+
+    if os.path.exists(action_entrypoint_file_path):
+        try:
+            os.remove(action_entrypoint_file_path)
+        except PermissionError:
+            LOG.error(
+                'No permission to delete the "%s" file',
+                action_entrypoint_file_path,
+            )
+            msg = 'No permission to delete "{0}" file from disk'.format(
+                action_entrypoint_file_path
+            )
+            abort(
+                http_client.INTERNAL_SERVER_ERROR,
+                six.text_type(msg),
+            )
```

Comment:
Please throw an exception here instead of doing a http abort.  Let the API controller do the abort for the http request on exception.